presidio-analyzer/presidio_analyzer/predefined_recognizers/phone_recognizer.py

```diff
@@ -66,52 +60,36 @@ def analyze(
         :return: List of phone numbers RecognizerResults
         """
         results = []
-        for entity in entities:
-            region = entity.replace(ENTITY_TYPE_SUFFIX, "")
-            if region in SUPPORTED_REGIONS or entity is INTERNATIONAL_ENTITY_TYPE:
-                for match in phonenumbers.PhoneNumberMatcher(text, region, leniency=0):
-                    international_phone_prefix = match.raw_string.startswith("+")
-                    if entity == INTERNATIONAL_ENTITY_TYPE \
-                            and international_phone_prefix:
-                        results += [self._get_international_recognizer_result(match)]
-                    # phone-numbers matches international numbers twice
-                    elif not international_phone_prefix:
-                        results += [
-                            self._get_regional_recognizer_result(
-                                match, entity, text, nlp_artifacts
-                            )
-                        ]
-
-        return results
-
-    def _get_regional_recognizer_result(self, match, entity, text, nlp_artifacts):
+        for region in self.supported_regions:
+            for match in phonenumbers.PhoneNumberMatcher(text, region, leniency=1):
+                results += [
+                    self._get_recognizer_result(match, text, region, nlp_artifacts)
+                ]
+
+        return EntityRecognizer.remove_duplicates(results)
+
+    def _get_recognizer_result(self, match, text, region, nlp_artifacts):
         number = match.number
-        main_region_code = COUNTRY_CODE_TO_REGION_CODE.get(number.country_code)[0]
         result = RecognizerResult(
-            entity_type=entity,
+            entity_type="PHONE_NUMBER",
             start=match.start,
             end=match.end,
             score=self.SCORE,
             analysis_explanation=self._get_analysis_explanation(),
```

Comment:
If this method gets the region as input, could we pass it to the analysis explanation somehow? I think it would be useful but it's not mandatory.

Reply:
That was the initial plan. However, since more than one region can be matched on the same number or as a substring, when removing duplicates we will lose that explanation. I tried to see if this can be added to the duplicates removal method but it was too complex to add.

Reply:
Even if we lose it, it's better than not having it at all in the first place, isn't it?

Reply:
It can reach a false explanation where you tried to find US numbers but the analysis stated that it matched only CA number, maybe we can phrase something differently?

Reply:
If you tried to find US numbers but it matched CA numbers, I think the user would benefit from knowing that. Users could change the supported regions if they want a more fine grain explanation. WDYT?

Reply:
Done